Controller/MediaAdminController.php

```diff
@@ -109,6 +117,116 @@ public function listAction(Request $request = null)
     }
 
     /**
+     * @throws \Symfony\Component\Security\Core\Exception\AccessDeniedException
+     *
+     * @return \Symfony\Bundle\FrameworkBundle\Controller\Response|\Symfony\Component\HttpFoundation\Response
+     */
+    public function multiUploadAction(Request $request)
+    {
+        if (false === $this->admin->isGranted('CREATE')) {
+            throw new AccessDeniedHttpException('Access denied');
+        }
+
+        $providerName = $this->getRequest()->get('provider');
+        $provider = $this->get($providerName);
+
+        // NEXT_MAJOR: remove $supportMethodCallable with the next major release.
+        $supportMethodCallable = method_exists($provider, 'supportsMultiUpload');
+
+        $defaultContext = $this->get('sonata.media.pool')->getDefaultContext();
+        $context = $this->get('request')->get('context', $defaultContext);
+
+        if (!$provider) {
+            $pool = $this->get('sonata.media.pool');
+            $providers = $pool->getProvidersByContext($context);
+
+            $filteredProviders = array();
+            /**
+             * @var $key string
+             * @var $provider MediaProviderInterface
```

Comment:
get ProvidersByContext should have a type hint: `@return MediaProviderInterface[]`